tests/e2e_logging.py

```diff
@@ -206,51 +206,64 @@ def test_protocols(network, args):
     url = f"{primary_root}/node/state"
     ca_path = os.path.join(network.common_dir, "service_cert.pem")
 
-    common_options = [url, "-sS", "--cacert", ca_path]
+    common_options = [
+        url,
+        "-sS",
+        "--cacert",
+        ca_path,
+        "-w",
+        "\\n%{http_code}\\n%{http_version}",
+    ]
 
-    # Check that websocket upgrade request is ignored
+    def parse_result_out(r):
+        assert r.returncode == 0, r.returncode
+        body = r.stdout.decode()
+        return body.rsplit("\n", 2)
+
+    # Call without any extra args to get golden response
     res = infra.proc.ccall(
         "curl",
-        "--no-buffer",
-        "-H",
-        "Connection: Upgrade",
-        "-H",
-        "Upgrade: websocket",
-        "-w",
-        "\n%{http_code}",
         *common_options,
     )
-    assert res.returncode == 0, res.returncode
-    body = res.stdout.decode()
-    status_code = body.splitlines()[-1]
-    assert status_code == "200", body
-    expected_response_body = body[: body.rfind("\n")]
-
-    # Test additional HTTP versions with curl
-    for (protocol, expected_errors) in (
-        ("", None),
-        ("--http1.0", None),
-        ("--http1.1", None),
-        ("--http2", None),  # Upgrade request is ignored
-        ("--http2-prior-knowledge", ["Error in the HTTP2 framing layer"]),
-        (
-            "--http3",
-            [
+    expected_response_body, status_code, http_version = parse_result_out(res)
+    assert status_code == "200", status_code
+    assert http_version == "1.1", http_version
+
+    # Test additional protocols with curl
+    for protocol, expected_result in {
```

Comment:
These test the same things, but now also confirm that the resulting HTTP version after negotiation is `1.1`, regardless of what we requested of curl. Additionally I swapped the "no args" and "try websockets" cases, so the former is now used to get the golden result.